pdr_backend/predictoor/predictoor_agent.py

```diff
@@ -481,22 +482,23 @@ def get_payout(self):
         pending_slots = query_pending_payouts(
             self.ppss.web3_pp.subgraph_url, up_pred_addr
         )
-        contracts = list(pending_slots.keys())
-        contracts_checksummed = [
-            self.ppss.web3_pp.web3_config.w3.to_checksum_address(addr)
-            for addr in contracts
-        ]
-        slots = list(pending_slots.values())
-        slots_flat = [item for sublist in slots for item in sublist]
-        slots_unique_flat = list(set(slots_flat))
-        if len(slots_unique_flat) == 0:
+        shared_slots = find_shared_slots(pending_slots)
+        if len(shared_slots) == 0:
             logger.info("No payouts available")
             return
 
-        tx = self.pred_submitter_mgr.get_payout(
-            slots_unique_flat, contracts_checksummed
-        )
-        logger.info("Payout tx: %s", tx["transactionHash"].hex())
+        for slot_tuple in shared_slots:
+            contract_addresses = slot_tuple[0]
```

Comment:
Replace 

```python
 contract_addresses = slot_tuple[0]
```

and (a few lines below)
```python
slots = slot_tuple[1]
```

with

```
contract_addresses, slots = slot_tuple
```

----

Also: replace `contract_addresses` -> `contract_addrs`

Why: always replace `address` -> `addr`, and `addresses` -> `addrs`, because it's just as descriptive / understandable, and more compact and readable.